Scripts/RemediationScripts/Remediate-DisableNonSSLPortOnRedisCache.ps1

```diff
@@ -388,46 +463,106 @@ function Disable-NonSSLPortOnRedisCache
                 if($RedisCacheResource.EnableNonSslPort -eq $false)
                 {
                     $RedisCacheRemediated += $RedisCache
+                    $logResource = @{}	
+                    $logResource.Add("ResourceGroupName",($_.ResourceGroupName))	
+                    $logResource.Add("ResourceName",($_.ResourceName))	
+                    $logRemediatedResources += $logResource	
                 }
                 else
                 {
                     $RedisCacheSkipped += $RedisCache
+                    $logResource = @{}	
+                    $logResource.Add("ResourceGroupName",($_.ResourceGroupName))	
+                    $logResource.Add("ResourceName",($_.ResourceName))
+                    $logResource.Add("Reason", "Error disabling Non-SSL port on Redis Cache(s): [$($RedisCache)]")               
+                    $logSkippedResources += $logResource	
                 }
             }
             catch
             {
                 $RedisCacheSkipped += $RedisCache
+                $logResource = @{}	
+                $logResource.Add("ResourceGroupName",($_.ResourceGroupName))	
+                $logResource.Add("ResourceName",($_.ResourceName))	
+                $logResource.Add("Reason","Encountered error while disabling Non-SSL port on Redis Cache(s) configuration")    	
+                $logSkippedResources += $logResource	
+                Write-Host "Skipping this resource..." -ForegroundColor $([Constants]::MessageType.Warning)	
```

Comment:
What is the user action needed in this case?

Reply:
this will require some research as this is on the control level and how it failed.